lib/vquic/ngtcp2.c

```diff
@@ -833,18 +903,51 @@ static int cb_h3_recv_header(nghttp3_conn *conn, int64_t stream_id,
      !memcmp(":status", h3name.base, h3name.len)) {
     int status = decode_status_code(h3val.base, h3val.len);
     DEBUGASSERT(status != -1);
-    msnprintf(stream->mem, stream->len, "HTTP/3 %03d \r\n", status);
+    /* make sure there's enough room for the status line, which is 13 chars */
```

Comment:
I could try to simplify this function if you think it's too long.

Reply:
I have no problems with it as-is, but if you can think of a nice way to split into a smaller one I'm all for that too! =)